docs/development/communication.md

```diff
@@ -1,11 +1,147 @@
 # Communication
 
-::: warning TODO
-Communication protocols with focus on energy management:
-- Low-layer
-    - CAN, RS485
-    - Wireless communication
-- Application layer
-    - Modbus, CANopen
-    - ThingSet protocol
-:::
+In a simple PV setup, one device might be sufficient to handle the panels, battery charging and load switching. But with a growing system, more and more device are added, with specific tasks for which they are optimized. A few [BMS](../system/bms.md) for the batteries, a couple of [charge controllers](../system/charge_controller.md) and maybe even a database and a frontend to store and display settings and data series.
+
+All these devices need to **communicate** with each other, with different requirements concerning speed, reliability or efficiency. For these network of devices (IoT = Internet of Things), several technologies have been developed. Similar to the [OSI model](https://en.wikipedia.org/wiki/OSI_model), different layers have been identified, but in this document we speak of the **application layer** and the **lower layers** which roughly corresponds to the data model and the data transportation.
+
+This chapter will give an overview of existing and popular protocols.
+
+## Requirements
+
+To be able to compare protocols, some specifications should be postulated. These are arbitrary based on the needs identified for Libre Solar devices and could be changed e.g. for automotive needs.
+
+### Lower layers
+
+1. **Reliability:** Paket loss, ordering of pakets for multiframe messages etc. should be handled. The connection should be somewhat resistant to noise since devices like the charge-controller use large conductances and high currents.
```

Comment:
```suggestion
1. **Reliability:** Packet loss, ordering of packets for multiframe messages etc. should be handled. The connection should be somewhat resistant to noise since devices like the charge-controller contain switching circuits and handle high currents.
```